tests/system/CLI/ConsoleTest.php

```diff
@@ -0,0 +1,81 @@
+<?php namespace CodeIgniter\CLI;
+
+use Config\MockCLIConfig;
+
+class ConsoleTest extends \CIUnitTestCase
+{
+
+	private $stream_filter;
+
+	public function setUp()
+	{
+		CLICTestStreamFilter::$buffer = '';
+		$this->stream_filter = stream_filter_append(STDOUT, 'CLICTestStreamFilter');
+
+		$this->env = new \CodeIgniter\Config\DotEnv(ROOTPATH);
+		$this->env->load();
+
+		// Set environment values that would otherwise stop the framework from functioning during tests.
+		if ( ! isset($_SERVER['app.baseURL']))
+		{
+			$_SERVER['app.baseURL'] = 'http://example.com';
+		}
+
+		$_SERVER['argv'] = ['spark', 'list'];
+		$_SERVER['argc'] = 2;
+		CLI::init();
+		
+		$this->app = new \CodeIgniter\MockCodeIgniter(new MockCLIConfig());
+//		$this->app->initialize();
+	}
+
+	public function tearDown()
+	{
+		stream_filter_remove($this->stream_filter);
+	}
+
+	public function testNew()
+	{
+		$console = new \CodeIgniter\CLI\Console($this->app);
+		$this->assertInstanceOf(Console::class, $console);
+	}
+
+	public function testHeader()
+	{
+		$console = new \CodeIgniter\CLI\Console($this->app);
+		$console->showHeader();
+		$result = CLICTestStreamFilter::$buffer;
+		$this->assertTrue(strpos($result, 'CodeIgniter CLI Tool') > 0);
+	}
+
+	public function testRun()
+	{
+		$console = new \CodeIgniter\CLI\Console($this->app);
+		$console->run();
+		$result = CLICTestStreamFilter::$buffer;
+
+		// make sure the result looks like a command list
+		$this->assertContains('Lists the available commands.', $result);
+		$this->assertContains('Displays basic usage information.', $result);
+	}
+
+}
+
+class CLICTestStreamFilter extends \php_user_filter
```

Comment:
Might consider moving this to its own class. Surely this isn't the only test that could benefit from this?

Reply:
I did that in one of the other PRs. I think we might need a separate such filter for each testing "package", because of the static variables.